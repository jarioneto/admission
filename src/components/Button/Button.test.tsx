import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { Button, ButtonProps } from './index';

const renderComponent = (props: ButtonProps = {}) =>
  render(<Button {...props}>Cancel</Button>);

describe('Button component', () => {
  it('should render the component', () => {
    renderComponent();

    expect(screen.getByRole('button')).toBeInTheDocument();
    expect(screen.getByText(/cancel/i)).toBeInTheDocument();
  });

  it('should trigger on click when pressed', async () => {
    const props = {
      onClick: jest.fn()
    };

    renderComponent(props);

    const button = screen.getByRole('button');
    await userEvent.click(button);

    expect(props.onClick).toHaveBeenCalled();
  });

  it('should do not trigger on click if disabled when pressed', async () => {
    const props = {
      disabled: true,
      onClick: jest.fn()
    };

    renderComponent(props);

    const button = screen.getByRole('button');
    await userEvent.click(button);

    expect(props.onClick).not.toHaveBeenCalled();
  });

  describe.each`
    variant
    ${'solid'}
    ${'outlined'}
  `('$variant', ({ variant }) => {
    it(`should render ${variant} button`, () => {
      const { container } = renderComponent({ variant });
      expect(container).toMatchSnapshot();
    });
  });

  describe.each`
    color
    ${'primary'}
    ${'secondary'}
    ${'tertiary'}
  `('$color', ({ color }) => {
    it(`should render ${color} button`, () => {
      const { container } = renderComponent({ color });
      expect(container).toMatchSnapshot();
    });
  });

  describe.each`
    shape
    ${'rectangle'}
    ${'circle'}
    ${'pill'}
  `('$shape', ({ shape }) => {
    it(`should render ${shape} button`, () => {
      const { container } = renderComponent({ shape });
      expect(container).toMatchSnapshot();
    });
  });
});