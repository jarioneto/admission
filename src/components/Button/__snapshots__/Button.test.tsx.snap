// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component circle should render circle button 1`] = `
<div>
  <button
    aria-disabled="false"
    class="sc-aYaIB YWqnm"
    color="primary"
    shape="circle"
    variant="outlined"
  >
    Cancel
  </button>
</div>
`;

exports[`Button component outlined should render outlined button 1`] = `
<div>
  <button
    aria-disabled="false"
    class="sc-aYaIB juLFoy"
    color="primary"
    shape="rectangle"
    variant="outlined"
  >
    Cancel
  </button>
</div>
`;

exports[`Button component pill should render pill button 1`] = `
<div>
  <button
    aria-disabled="false"
    class="sc-aYaIB jFtxKj"
    color="primary"
    shape="pill"
    variant="outlined"
  >
    Cancel
  </button>
</div>
`;

exports[`Button component primary should render primary button 1`] = `
<div>
  <button
    aria-disabled="false"
    class="sc-aYaIB juLFoy"
    color="primary"
    shape="rectangle"
    variant="outlined"
  >
    Cancel
  </button>
</div>
`;

exports[`Button component rectangle should render rectangle button 1`] = `
<div>
  <button
    aria-disabled="false"
    class="sc-aYaIB juLFoy"
    color="primary"
    shape="rectangle"
    variant="outlined"
  >
    Cancel
  </button>
</div>
`;

exports[`Button component secondary should render secondary button 1`] = `
<div>
  <button
    aria-disabled="false"
    class="sc-aYaIB bCUTGq"
    color="secondary"
    shape="rectangle"
    variant="outlined"
  >
    Cancel
  </button>
</div>
`;

exports[`Button component solid should render solid button 1`] = `
<div>
  <button
    aria-disabled="false"
    class="sc-aYaIB iWBPOX"
    color="primary"
    shape="rectangle"
    variant="solid"
  >
    Cancel
  </button>
</div>
`;

exports[`Button component tertiary should render tertiary button 1`] = `
<div>
  <button
    aria-disabled="false"
    class="sc-aYaIB bgcTgy"
    color="tertiary"
    shape="rectangle"
    variant="outlined"
  >
    Cancel
  </button>
</div>
`;
